app/src/main/kotlin/com/simplemobiletools/flashlight/activities/MainActivity.kt

```diff
@@ -1,207 +1,250 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.res.stringResource
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewModelScope
+import com.google.android.material.math.MathUtils
+import com.simplemobiletools.commons.compose.alert_dialog.AlertDialogState
+import com.simplemobiletools.commons.compose.alert_dialog.rememberAlertDialogState
+import com.simplemobiletools.commons.compose.extensions.CheckAppOnSdCard
+import com.simplemobiletools.commons.compose.extensions.onEventValue
+import com.simplemobiletools.commons.compose.theme.AppThemeSurface
+import com.simplemobiletools.commons.dialogs.ConfirmationDialog
+import com.simplemobiletools.commons.dialogs.PermissionRequiredAlertDialog
+import com.simplemobiletools.commons.dialogs.RadioGroupAlertDialog
 import com.simplemobiletools.commons.extensions.*
 import com.simplemobiletools.commons.helpers.*
 import com.simplemobiletools.commons.models.FAQItem
 import com.simplemobiletools.commons.models.RadioItem
 import com.simplemobiletools.flashlight.BuildConfig
 import com.simplemobiletools.flashlight.R
-import com.simplemobiletools.flashlight.databinding.ActivityMainBinding
-import com.simplemobiletools.flashlight.dialogs.SleepTimerCustomDialog
+import com.simplemobiletools.flashlight.dialogs.SleepTimerCustomAlertDialog
 import com.simplemobiletools.flashlight.extensions.config
+import com.simplemobiletools.flashlight.extensions.startAboutActivity
 import com.simplemobiletools.flashlight.helpers.*
-import com.simplemobiletools.flashlight.models.Events
-import org.greenrobot.eventbus.EventBus
-import org.greenrobot.eventbus.Subscribe
-import org.greenrobot.eventbus.ThreadMode
+import com.simplemobiletools.flashlight.screens.*
+import com.simplemobiletools.flashlight.views.AnimatedSleepTimer
+import kotlinx.collections.immutable.toImmutableList
+import kotlinx.coroutines.flow.*
 import java.util.*
+import kotlin.system.exitProcess
 
-class MainActivity : SimpleActivity() {
+class MainActivity : ComponentActivity() {
     companion object {
         private const val MAX_STROBO_DELAY = 2000L
         private const val MIN_STROBO_DELAY = 10L
-        private const val FLASHLIGHT_STATE = "flashlight_state"
-        private const val STROBOSCOPE_STATE = "stroboscope_state"
     }
 
-    private val binding by viewBinding(ActivityMainBinding::inflate)
-
-    private var mBus: EventBus? = null
-    private var mCameraImpl: MyCameraImpl? = null
-    private var mIsFlashlightOn = false
-    private var reTurnFlashlightOn = true
+    private val preferences by lazy { config }
+    private val viewModel by viewModels<MainViewModel>()
 
     override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
Main activity is not edge to edge

Reply:
Is it supposed to be? Since it wasn't before the migration either.

Reply:
It's there, install from play store and see.

Reply:
Yeah, my bad. I will add it.